elements/drag-drop-elements/pl-drag-drop-element.py

```diff
@@ -0,0 +1,389 @@
+            # BACKWARD COMPATIBILITY CODE
+            if html_tags.tag == 'pl-distractor':
+                mcq_options.append(str.strip(html_tags.text))   # store the original specified ordering of all the MCQ options
+
+        answerName = pl.get_string_attrib(element, 'answers-name')
+        header_left_column = pl.get_string_attrib(element, 'header-left-column', 'Drag from here:')
+        header_right_column = pl.get_string_attrib(element, 'header-right-column', 'Construct your solution here:')
+
+        student_submission_dict_list = []
+
+        mcq_options = data['params'][answerName]
+
+        if answerName in data['submitted_answers']:
+            student_previous_submission = data['submitted_answers'][answerName]['student_raw_submission']
+            mcq_options = list(set(mcq_options) - set(student_previous_submission))
+
+        for index, mcq_options_text in enumerate(student_previous_submission):
+            # render the answers column (restore the student submission)
+            submission_indent = data['submitted_answers'][answerName]['student_answer_indent'][index]
+            submission_indent = (int(submission_indent) * 50) + 5
+            temp = {'text': mcq_options_text, 'indent': submission_indent}
+            student_submission_dict_list.append(dict(temp))
+
+        html_params = {
+            'question': True,
+            'answerName': answerName,
+            'options': mcq_options,
+            'header-left-column': header_left_column,
+            'header-right-column': header_right_column,
+            'submission_dict': student_submission_dict_list
+            # 'data': data
+        }
+
+        with open('pl-drag-drop-element.mustache', 'r', encoding='utf-8') as f:
+            html = chevron.render(f, html_params).strip()
+        return html
+
+    elif data['panel'] == 'submission':
+        if pl.get_boolean_attrib(element, 'external-grader', False):  # if True
+            return ''
+        # render the submission panel
+        uuid = pl.get_uuid()
+        student_submission = ''
+        colour = 'badge-danger'
+        score = 0
+        feedback = None
+
+        if answerName in data['submitted_answers']:
+            student_submission = data['submitted_answers'][answerName]['student_raw_submission']
+        if answerName in data['partial_scores']:
+            colour = render_html_colour(data['partial_scores'][answerName]['score'])
+            score = data['partial_scores'][answerName]['score'] * 100
+            feedback = data['partial_scores'][answerName]['feedback']
+
+        html_params = {
+            'submission': True,
+            'uuid': uuid,
+            'parse-error': data['format_errors'].get(answerName, None),
+            'student_submission': student_submission,
+            'colour': colour,
+            'score': score,
+            'perfect_score': True if score == 100 else None,
+            'feedback': feedback
+            # 'data': data
```

Comment:
Shouldn't have commented out code in a pull request.

Reply:
fixed in bcadc9f